### ISSUE TITLE
Simplified general purpose filtering system

### ISSUE THREAD

**Issue by Deathspike** [Deathspike/mangarack#53]
I was thinking about the filtering options as proposed by @murraypaul in #51. I've always had a bit of an issue with the various filtering options that we've had in the `3.x` branch, and I've had the same issues with the proposed filters in that particular pull request. There are many of them, and they aren't particularly clear. It just hit me; **we're thinking about solutions and not about the problem**.

The underlying problem is that *we are downloading chapters that we do not want*. Thus, we all come up with various methods to reduce the chapter list based on various conditions. But all we really want to do is *not download chapters we consider to have seen*. **We can just make a complete list of chapters to exclude**. 

If you remember `3.x`, it had an `.mrignore` (or something along those lines) which tracked files that had been downloaded so that you could move or delete chapters you had downloaded once. This exact same list can be employed for filtering purposes. **We just need to generate one**. So

1. Run `mangarack http://mangafox.me/manga/futari_dake/ --createExclusions` to create exclusion file.
2. (Optional) Edit the exclusion file to remove chapters you *do* want to download.
3. Run ``mangarack http://mangafox.me/manga/futari_dake/`. Exclusions are now respected.

New files come in. This removes the entire need for filtering. Thoughts? (cc @XellossMetallium)

**Comment by murraypaul** [comment]
Some thoughts.
- To a certain extent this is just pushing the filtering issue somewhere else, as the users still have to determine which lines to delete. However, as long as all of the information from the site is transferred to the exclusion file, this should be possible, and there are lots of existing tools for searching/replacing/deleting lines from a file.
- What happens the next time you want to download? Do you create exclusions again, and is the program able to only add chapters that have appeared since the last time the file was created, and not add back in all the lines the user has already deleted? Or would it not matter, because the files would already have been downloaded, so it is fine to exclude them next time?
- Would the problem above be easier if mangarack did have an uploaded-since filter?

**Comment by murraypaul** [comment]
The alternative approach is to add so much information to the downloaded filename so that **all** chapters can be downloaded, and the user can 'filter' the downloaded files, rather than mangarack having to deal with it at all.

**Comment by Deathspike** [comment]
Exactly. You're just moving the filtering elsewhere. Except that a simple file on the file system that is used as an exclusion list is quite transparent. You don't have to deal with any kind of chapter/volume/group filtering with any kind of command line flag. Essentially, each line in the exclusion file is a *virtual comic book*. MangaRack will think it exist and skip downloading it. That also means that new files that don't exist as file or exclusion will always be downloaded, which is exactly what is expected.

Concerning all metadata in the filename; that is another discussion about metadata and ComicRack.

**Comment by Maximilian-N** [comment]
This really sounds easy and predictable. You just create and edit your file once and forget about it. And there also seems to be almost no way for this list to interact with other parts of mangarack in an undesired way.

**Comment by murraypaul** [comment]
Pondering this a bit more, the various filters I added addressed two different problems, of which this is the first:
a) We are downloading chapters we don't want
b) We are downloading the wrong files for chapters we do want
This solves a, b is a separate issue, caused by a disconnect between Batatao, which operates on files, and can have multiple files for a single chapter, and mangarack, which operates on chapters, and always produces a single file for a chapter. b can be discussed separately.

**Comment by XellossMetallium** [comment]
As I always said I do not particullary mind saving drive space.. my concerns are more about what I miss (see #2 ), so generally speaking I'd prefer duplicates over the risk of missing something. Obviously having both would be great, but for that very reason I do not miss filters so much (the excluding ones). By the way my ideal setup would consist in a nas running MangaRack periodically according to batoto reading list, a MR Library desktop and android app to read via lan or sync to device and a magic button to convert to epub and send to epub reader. :D

Going back on earth: I also thought about possible solutions and actually came up with something similar to what @Deathspike introduced here.. to be precise something like "dummy" chapters, like empty placeholders renamed after not needed chapters.  
Obviously a single list is cleaner and easier to manage :D 
All things considered I do like this solution, it seems a good compromise, excluding everything that would substantially mean to return to v3.
